doc/contrib/docs.rst

```diff
@@ -8,23 +8,73 @@ Documentation and Examples
   :backlinks: none
   :local:
 
-*********
-Documents
-*********
+*************
+Documentation
+*************
 * Python and C documentation is built using `Sphinx <http://www.sphinx-doc.org/en/master/>`_.
 * Each document is written in `reStructuredText <http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_.
-* You can build document locally to see the effect, by running
+* The documentation is the ``doc/`` directory.
+* You can build it locally using ``make html`` command.
 
-.. code-block:: bash
+   .. code-block:: bash
 
-  make html
+     make html
 
-inside the ``doc/`` directory.  The online document is hosted by `Read the Docs <https://readthedocs.org/>`__ where the imported project is managed by `Hyunsu Cho <https://github.com/hcho3>`__ and `Jiaming Yuan <https://github.com/trivialfis>`__.
+The online document is hosted by `Read the Docs <https://readthedocs.org/>`__ where the imported project is managed by `Hyunsu Cho <https://github.com/hcho3>`__ and `Jiaming Yuan <https://github.com/trivialfis>`__.
+
+===============================
+Build Docs on macOS using Conda
+===============================
+
+#. Create a conda environment.
+
+   .. code-block:: bash
+
+     conda create -n xgboost-docs --yes python=3.10
+
+   .. note:: Python 3.10 is required by `xgboost_ray <https://github.com/ray-project/xgboost_ray>`__ package.
+
+#. Activate the environment
+
+   .. code-block:: bash
+
+     conda activate xgboost-docs
+
+#. Install required packages (in the current environment) using ``pip`` command.
+   For some reason, it is currently not possible to install the required packages using ``conda``.
```

Comment:
I installed the required packages from conda with packages from the `conda-forge` channel.

Would be great to keep the toolchain as consistent as possible.

Reply:
The issue I had was with ray and Python version. I started with Python 3.11 and `make html` couldn't resolve xgboost-ray. Once I switched to Python 3.10 it all started well but only with `pip`. I couldn't convince `conda` to resolve all the deps. I'll give it a try again and report back with the findings.

Reply:
This is the issue I'm facing with `conda install`. Any idea how to fix it (and how it's possible that it works for you?!)

``` console
$ conda install --file requirements.txt --yes -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - xgboost_ray

Current channels:

  - https://conda.anaconda.org/conda-forge/osx-64
  - https://conda.anaconda.org/conda-forge/noarch
  - https://repo.anaconda.com/pkgs/main/osx-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/osx-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```